codetransformer/core.py

```diff
@@ -16,15 +18,29 @@ def _a_if_not_none(a, b):
     return a if a is not None else b
 
 
-class NoCodeContext(Exception):
-    """Exection raised to indicate that the ``code`` attribute was accessed
-    outside of a code context.
+class NoContext(Exception):
+    """Exception raised to indicate that the ``code` or ``startcode``
+    attribute was accessed outside of a code context.
     """
     def __init__(self):
-        return super().__init__('no code context')
+        return super().__init__('no context')
 
 
-class CodeTransformer(object):
+class CodeTransformerMeta(type):
+    """Meta class for CodeTransformer to collect all pf the patterns
```

Comment:
Can we document where these patterns actually come from (e.g. note that we're collection methods decorated in a certain way)?